crates/net/network/src/eth_requests.rs

```diff
@@ -201,26 +201,28 @@ where
         response: oneshot::Sender<RequestResult<Receipts>>,
     ) {
         let mut receipts = Vec::new();
-
-        let mut total_bytes = APPROX_RECEIPT_SIZE;
+        let mut total_len = 0;
 
         for hash in request.0 {
             if let Some(receipts_by_block) =
                 self.client.receipts_by_block(BlockHashOrNumber::Hash(hash)).unwrap_or_default()
             {
-                receipts.push(
-                    receipts_by_block
-                        .into_iter()
-                        .map(|receipt| receipt.with_bloom())
-                        .collect::<Vec<_>>(),
-                );
+                let block_receipts = receipts_by_block
+                    .into_iter()
+                    .map(|receipt| receipt.with_bloom())
+                    .collect::<Vec<_>>();
 
-                total_bytes += APPROX_RECEIPT_SIZE;
+                let mut buf = BytesMut::new();
+                block_receipts.encode(&mut buf);
```

Comment:
now we determine the correct limit by encoding the receipts, however we're doing the work twice because we're encoding here but still send the unencoded receipts, which means we will do the work again.

what we want is a solution like in #2246 which supports sending encoding txs as the response